pkg/config/init.go

```diff
@@ -1,105 +1,105 @@
+	HOSTNAME          = os.Getenv("HOSTNAME")
+	LOCAL_ADMIN_URL   = ""
+	podInformer       coreinformers.PodInformer
+	svcInformer       coreinformers.ServiceInformer
+	nsInformer        coreinformers.NamespaceInformer
 	EndpointsInformer coreinformers.EndpointsInformer
-	IsLeader = false
+	IsLeader          = false
 	//etcdClient client.Client
-	kubeClient kubernetes.Interface
+	kubeClient                kubernetes.Interface
 	CoreSharedInformerFactory informers.SharedInformerFactory
 )
+
 const PROD = "prod"
 const HBPROD = "hb-prod"
 const BETA = "beta"
 const DEV = "dev"
 const TEST = "test"
 const LOCAL = "local"
-const confPath = "/root/ingress-controller/conf.json"
 const AispeechUpstreamKey = "/apisix/customer/upstream/map"
 
-func setEnvironment() {
-	if env := os.Getenv("ENV"); env == "" {
-		ENV = LOCAL
-	} else {
-		ENV = env
-	}
-	_, basePath, _, _ = runtime.Caller(1)
-}
+// Config contains necessary config items for running apisix-ingress-controller.
+type Config struct {
+	Hostname string `json:"-"`
 
-func ConfPath() string {
-	if ENV == LOCAL {
-		return filepath.Join(filepath.Dir(basePath), "conf.json")
-	} else {
-		return confPath
-	}
+	SyslogServer string       `json:"syslog_server"`
+	Kubeconfig   string       `json:"kubeconfig"`
+	Etcd         EtcdConfig   `json:"etcd"`
+	APISIX       APISIXConfig `json:"apisix"`
 }
 
-type etcdConfig struct {
-	Addresses []string
+// EtcdConfig contains config items about etcd.
+type EtcdConfig struct {
+	Endpoints []string `json:"endpoints"`
 }
 
-var EtcdConfig etcdConfig
-var K8sAuth k8sAuth
-var Syslog syslog
-
-var config *restclient.Config
+// APISIXConfig contains config items about apisix.
+type APISIXConfig struct {
+	BaseURL string `json:"base_url"`
+}
 
-func init() {
-	// 获取当前环境
-	setEnvironment()
-	// 获取配置文件路径
-	filePath := ConfPath()
-	// 获取配置文件内容
-	if configurationContent, err := ioutil.ReadFile(filePath); err != nil {
-		panic(fmt.Sprintf("failed to read configuration file: %s", filePath))
-	} else {
-		configuration := gjson.ParseBytes(configurationContent)
-		// apisix baseUrl
-		apisixConf := configuration.Get("conf.apisix")
-		apisixBaseUrl := apisixConf.Get("base_url").String()
-		seven.SetBaseUrl(apisixBaseUrl)
-		// k8sAuth conf
-		k8sAuthConf := configuration.Get("conf.k8sAuth")
-		K8sAuth.file = k8sAuthConf.Get("file").String()
-		// syslog conf
-		syslogConf := configuration.Get("conf.syslog")
-		Syslog.Host = syslogConf.Get("host").String()
+// NewDefaultConfig creates a Config object filled by default value.
+func NewDefaultConfig() *Config {
+	return &Config{
+		Hostname: _hostname,
 	}
-	// init etcd client
-	//etcdClient = NewEtcdClient()
-	// init informer
-	InitInformer()
 }
 
-type k8sAuth struct {
-	file string
+// NewConfigFromFiles creates a Config object and fill it by values in configuration file.
+func NewConfigFromFile(configPath string) (*Config, error) {
+	data, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		return nil, err
```

Comment:
I think we need to capture the error and write it to the error log file.

we can do this in a new PR. please create a new issue about this.